src/core/CMakeLists.txt

```diff
@@ -156,15 +156,21 @@ if (WITH_CUDA)
     cuda/cuda_compilation_cache.cc
     cuda/cuda_rtc.cc
     cuda/cuda_tc_executor.cc
+    cuda/cuda_profile.cc
   )
   target_include_directories(tc_cuda PUBLIC ${PROJECT_SOURCE_DIR}/include ${LLVM_INCLUDE_DIRS})
+  target_include_directories(tc_cuda PUBLIC 
+      ${PROJECT_SOURCE_DIR}/include
+      ${LLVM_INCLUDE_DIRS} 
+      ${CUDA_TOOLKIT_ROOT_DIR}/extras/CUPTI/include)
   target_link_libraries(
     tc_cuda
 
     ${CUDA_CUDA_LIBRARIES}
     ${CUDA_curand_LIBRARY}
     ${CUDA_LIBRARIES}
     ${CUDA_NVRTC_LIBRARIES}
+    ${CUDA_cupti_LIBRARY}
```

Comment:
`CUDA_cupti_LIBRARY` doesn't seem to be defined anywhere.

I get
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_cupti_LIBRARY (ADVANCED)
    linked by target "tc_cuda" in directory /home/skimo/git/c2isl/src/core
```

Reply:
`find_package(cuda)` should have set it. Is there a cupti directory under `cuda_root_dir/extras` in your machine?

Reply:
You shouldn't have to install it separately (I didn't).
```
The CUPTI library is supported on all platforms supported by the CUDA Toolkit, and is available on the CUDA Downloads page as part of the CUDA Tools SDK.
```